docs/data-model.rst

```diff
@@ -384,6 +385,9 @@ The ``site`` column is an integer value defining the ID of the
 The ``node`` column is an integer value defining the ID of the
 first :ref:`node <sec_node_table_definition>` in the tree below this mutation.
 
+The  ``time`` column is a double precision floating point value recording when the
+mutation happened. This is the same time dimension as that for nodes.
```

Comment:
How about just "recording how long ago the mutation happened", without the reference to the node's "time dimension"?

Reply:
Fixed in https://github.com/tskit-dev/tskit/pull/672/commits/6296a8cee08b755e0fba43390e144bd96768c23e